java/ext/javadoctest.py

```diff
@@ -54,6 +54,7 @@ def compile(
                 "dependency:build-classpath",
                 "-DincludeTypes=jar",
                 "-Dmdep.outputFile=.cp.tmp",
+                f"-Darrow.version={self.env.config.version}",
```

Comment:
For nightly version, this will be obtained from the last build version at nightlies.apache.org, could be needed to be a logic to obtain last version and use thata

Reply:
this gets read from the `conf.py` value. If it's nightly it will always be `10.0.0-SNAPSHOT` (or other X.0.0-SNAPSHOT version for future releases) with the changes  we have done to how we publish the version for the JAVA nightlies. The logic to select this version is on `java/source/conf.py`